src/modules/paymentProcessor/PP_Queue_v1.sol

```diff
@@ -0,0 +1,740 @@
+    function processPayments(IERC20PaymentClientBase_v1 client_)
+        external
+        virtual
+        clientIsValid(address(client_))
+        onlyModule
+    {
+        // Collect outstanding orders and their total token amount.
+        IERC20PaymentClientBase_v1.PaymentOrder[] memory orders;
+
+        (orders,,) = client_.collectPaymentOrders();
+
+        uint orderLength = orders.length;
+        for (uint i; i < orderLength; ++i) {
+            // Add order to order queue
+            _addPaymentOrderToQueue(orders[i], address(client_));
+        }
+        // Execute Order Queue.
+        _executePaymentQueue(address(client_));
+    }
+
+    /// @inheritdoc IPaymentProcessor_v1
+    function cancelRunningPayments(IERC20PaymentClientBase_v1 client_)
+        external
+        view
+        clientIsValid(address(client_))
+    {
+        return;
+    }
+
+    /// @inheritdoc IPaymentProcessor_v1
+    function unclaimable(address client, address token, address paymentReceiver)
+        public
+        view
+        returns (uint amount_)
+    {
+        amount_ =
+            _unclaimableAmountsForRecipient[client][token][paymentReceiver];
+    }
+
+    /// @inheritdoc IPaymentProcessor_v1
+    function claimPreviouslyUnclaimable(
+        address client,
+        address token,
+        address receiver
+    ) external {
+        if (unclaimable(client, token, _msgSender()) == 0) {
+            revert Module__PaymentProcessor__NothingToClaim(
+                client, _msgSender()
+            );
+        }
+
+        _claimPreviouslyUnclaimable(client, token, receiver);
+    }
+
+    /// @inheritdoc IPaymentProcessor_v1
+    function validPaymentOrder(
+        IERC20PaymentClientBase_v1.PaymentOrder memory order_
+    ) external view returns (bool isValid_) {
+        // Extract queue ID from order data.
+        uint queueId_ = _getPaymentQueueId(order_.flags, order_.data);
+
+        // Validate payment receiver, amount and queue ID.
+        isValid_ = _validPaymentReceiver(order_.recipient)
+            && _validTotalAmount(order_.amount) && _validQueueId(queueId_)
+            && _validPaymentToken(order_.paymentToken);
+    }
+
+    /// @inheritdoc IPP_Queue_v1
+    function cancelPaymentOrderThroughQueueId(uint orderId_)
+        external
+        onlyModuleRole(QUEUE_OPERATOR_ROLE)
+        returns (bool success_)
+    {
+        // Validate queue ID.
+        if (!_orderExists(orderId_)) {
+            revert Module__PP_Queue_InvalidOrderId(
+                _orders[orderId_].client, orderId_
+            );
+        }
+
+        QueuedOrder storage order = _orders[orderId_];
+
+        // Check if order can be cancelled
+        if (order.state != RedemptionState.PROCESSING) {
+            revert Module__PP_Queue_InvalidState();
+        }
+
+        // Update order state and emit event
+        _updateOrderState(orderId_, RedemptionState.CANCELLED);
+
+        // Add amount to unclaimable
+        _setUnclaimableAmount(
```

Comment:
This is overriding the currently stored amount for that user (for that token + client). Should this be added or is it intentional to fully override it, even in case the user has multiple orders?

Reply:
hey good catch! You're right. I've modified it to add on top of the existing amount and renamed function to `_addToUnclaimableAmount`